pkg/proxy/validate.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright © 2023 The Spray Proxy Contributors
+
+SPDX-License-Identifier: Apache-2.0
+*/
+package proxy
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/google/go-github/v51/github"
+)
+
+const envWebhookSecret = "GH_APP_WEBHOOK_SECRET"
+
+// calls validation functions on request
+func validateProxyRequest(req *http.Request, insecureWebhook bool) error {
+	if err := validateWebhookSignature(req, insecureWebhook); err != nil {
+		return err
+	}
+	return nil
+}
+
+func validateWebhookSignature(req *http.Request, insecureWebhook bool) error {
+	if !insecureWebhook {
+		if secret := os.Getenv(envWebhookSecret); secret != "" {
```

Comment:
nit (non-blocking) - the secret should be passed into this function as a parameter. This makes code easier to unit test.

I consider best practice to read configuration options (like env vars) as close to the main entrypoint as possible, and then pass those down to the "business logic" through struct objects and function arguments.

Reply:
That was my initial implementation, specifically with the unit testing in mind. Then I was thrown back by the increasing number of parameters passed around and specifically at NewServer and NewSprayProxy. Anyway, that makes sense and I'll refactor tomorrow.
Now I also pay attention that I diverged from what we agreed on the ticket, that the server should panic if the secret is not found and here it only affects the specific request. I'll fix that as well.

Reply:
I made the secret param of validateWebhookSignature.
Management of the secret env var was moved as "early" as the start of NewSprayProxy. The current implementation requires both insecureWebhook config and the actual secret and I did not want to pass 2 extra variables from "main" -> NewServer -> NewSprayProxy.
Having the validation enabled (default) and missing the secret now makes the server to exit.